lib/galaxy/tool_util/toolbox/base.py

```diff
@@ -882,11 +882,8 @@ def _load_tool_tag_set(
             if labels is not None:
                 tool.labels = labels
         except OSError as exc:
-            msg = "Error reading tool configuration file from path '%s': %s", path, unicodify(exc)
-            if exc.errno == ENOENT:
-                log.error(msg)
-            else:
-                log.exception(msg)
+            exc_info = exc.errno != ENOENT
+            log.error("Error reading tool configuration file from path '%s': %s", path, exc, exc_info=exc_info)
```

Comment:
Should this be just `log.error("Error reading tool configuration file from path '%s'", path, exc_info=exc_info)`?

Reply:
And why is the unicodify not needed anymore?

Reply:
If you remove `exc`, then you won't have any info about the exception when `exc_info` is False, so I think the small repetition is better.
I don't think `unicodify()` was needed in the first place, it may be a leftover from the Python2 days.